05-specifying-sample-designs.Rmd

```diff
@@ -1 +1,673 @@
+
+des_acs
+
+summary(des_acs)
+```
+
+
+### Bootstrap method
+
+In bootstrap resampling, replicates are created by selecting random samples of the PSUs with replacement. If there are $M$ PSUs in the sample, then each replicate will be created by selecting a random sample of $M$ PSUs with replacement. Each replicate is created independently and the weights for each replicate are adjusted to to reflect the population generally using the same method as the analysis weight was adjusted. 
+
+#### The math {-}
+
+A weighted estimate for the full sample is calculated as $\hat{\theta}$ and then a weighted estimate for each replicate is calculated as $\hat{\theta}_r$ for $R$ replicates. Then the standard error of the estimate is calculated as:
+
+$$se(\hat{\theta})=\sqrt{\alpha \sum_{r=1}^R \left( \hat{\theta}_r-\hat{\theta}\right)^2}$$
+
+#### The syntax {-}
+
+If a dataset had WT0 for the main weight, had 20 bootstrap weights indicated WT1, WT2, ..., WT20, and $\alpha=.02$, use the following syntax:
+
+```r
+des_bs <- dat %>%
+  as_survey_rep(weights = WT0, repweights= num_range("WT", 1:20),
+                type="bootstrap", mse=TRUE, scale=.02)
+
+```
+
+Note that the scale is usually provided in documentation and is a constant so is not provided as a variable in the tibble.
+
+#### Example {-}
+
+Returning to the api example, bootstrap weights were constructed for a one cluster design. 50 replicate weights were created on a dataset `apiclus1_slim` which has some familiar variables including cds, dnum, fpc, and pw but now additionally includes bootstrap weights pw1, ..., pw50. The scale $(\alpha)$ is $15/(14*49)=0.02186589$
+
+```{r}
+#| label: genbs
+#| include: FALSE
+apiclus1_slim <- 
+  apiclus1 %>%
+  as_tibble() %>%
+  arrange(dnum) %>%
+  select(cds, dnum, fpc, pw)
+
+set.seed(662152)
+apibw <- bootweights(psu=apiclus1_slim$dnum, strata=rep(1, nrow(apiclus1_slim)), fpc=apiclus1_slim$fpc, replicates=50)
+
+bwmata <- apibw$repweights$weights[apibw$repweights$index, ]*apiclus1_slim$pw
+
+apiclus1_slim <- bwmata %>%
+  as.data.frame() %>%
+  set_names(str_c("pw", 1:50)) %>%
+  cbind(apiclus1_slim) %>%
+  as_tibble() %>%
+  select(cds, dnum, fpc, pw, everything())
+```
+
+```{r}
+#| label: bsexamp
+apiclus1_slim
+
+des_api1_bs <- apiclus1_slim %>% 
+  as_survey_rep(weights=pw, repweights=pw1:pw50, type="bootstrap", 
+                scale=0.02186589, mse=TRUE)
+
+des_api1_bs
+
+summary(des_api1_bs)
+```
+
+As with other replicate design objects, when printing the object or looking at the summary, the replicate weights are provided along with the data variables.
+
+## Understanding survey design documentation
+
+SRS, stratified, and clustered designs are the backbone of sampling designs and the features are often combined in one design. Additionally, rather than using SRS for selection, other mechanisms are commonly used such as probability proportional to size (PPS), systematic sampling, or selection with unequal probabilities.
```

Comment:
It would probably be good to add a reference to a resource for identifying what PPS, etc is.  May want to add one sentence here as well since you're saying "other mechanisms" and be more specific.  Yes they are other mechanisms, but more clearly they are sampling strategies.